lib/content._coffee

```diff
@@ -47,11 +44,14 @@ getFilePathForId = (id) ->
     id = id.replace /([A-Z])/g, '_$1'
     Path.join DIR_BY_ID_PATH, "#{id}.json"
 
-getFilePathForURL = (url) ->
-    if idFile = DATABASE["#{hashURL url}.json"]
-        Path.join DIR_BY_ID_PATH, idFile.replace '../content-by-id', ''
-    else
-        ''
+readFileForURL = (url, _) ->
+    cloudClient.download
+        container: 'content/content-by-url'
+        remote: "#{hashURL url}.txt"
+
+    # Somehow after the download we get the ID and then ...
```

Comment:
Yep, `download` returns a [readable stream](http://nodejs.org/dist/v0.10.36/docs/api/stream.html#stream_api_for_stream_consumers). Readable streams emit `end` (success) or `error` (failure) events, with `data` events beforehand for whatever's been read.

It looks like `cloudClient.download` *also* takes a callback... but it looks like it only passes the metadata for the file, not the file itself. It supports downloading the file to disk, but we probably don't want that for these tiny `id` files.

So we have to implement the logic for piecing together the `data` events, like that stream documentation example shows. And then we'll listen for `end` and `error` to either resume or fail, like here:

https://github.com/zoomhub/zoomhub/blob/cloud-files/lib/worker._coffee#L212-L213